projects/igniteui-angular/src/lib/directives/button/button.directive.ts

```diff
@@ -53,22 +54,16 @@ export class IgxButtonDirective extends DisplayDensityBase {
     private static ngAcceptInputType_type: IgxButtonType | '';
 
     /**
-     * Gets or sets whether the button is selected.
-     * Mainly used in the IgxButtonGroup component and it will have no effect if set separately.
-     *
-     * @example
-     * ```html
-     * <button igxButton="flat" [selected]="button.selected"></button>
-     * ```
+     * Called when the button is clicked.
      */
-    @Input()
-    public selected = false;
+    @Output()
+    public buttonClick = new EventEmitter<any>();
```

Comment:
Should this be event emitter of `any` type?

Reply:
Perhaps not, but fixing the click event is not the subject of this PR.